Add controllers

IgorMartynkin1981 opened the pull request:
Refactor согласно ТЗ_13

IgorMartynkin1981 committed:
Refactor User

IgorMartynkin1981 committed:
Refactor User

IgorMartynkin1981 committed:
Refactor User

IgorMartynkin1981 committed:
Refactor add Item cod

kukolev commented on src/main/java/ru/practicum/shareit/item/Item.java:
```diff
@@ -0,0 +1,19 @@
+import lombok.Data;
+
+/**
+ * TODO Sprint add-controllers.
```
Это TODO нужно удалить, т.к. функционал реализован. Либо нужно написать что осталось реализовать.

IgorMartynkin1981 replied:
Заметки убрал

kukolev commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,53 @@
+import java.util.Collection;
 
 /**
  * TODO Sprint add-controllers.
```
Это TODO нужно удалить, т.к. функционал реализован. Либо нужно написать что осталось реализовать.

IgorMartynkin1981 replied:
Заметки убрал

kukolev commented on src/main/java/ru/practicum/shareit/item/ItemDto.java:
```diff
@@ -0,0 +1,18 @@
+import lombok.Data;
+
+/**
+ * TODO Sprint add-controllers.
```
Это TODO нужно удалить, т.к. функционал реализован. Либо нужно написать что осталось реализовать.

IgorMartynkin1981 replied:
Заметки убрал

kukolev commented on src/main/java/ru/practicum/shareit/item/MemoryItemDAOImpl.java:
```diff
@@ -0,0 +1,67 @@
+            items.put(itemId, item);
+            return item;
+        } else {
+            throw new StorageException(String.format("There are no any items with ID %d!", itemId), HttpStatus.NOT_FOUND);
```
Нарушение требований кодстайла: Очень длинная строка - превышает 120 символов в длину.

IgorMartynkin1981 replied:
Выполнил перенос строки

kukolev commented on src/main/java/ru/practicum/shareit/item/MemoryItemDAOImpl.java:
```diff
@@ -0,0 +1,67 @@
+        if (items.containsKey(itemId)) {
+            return items.get(itemId);
+        } else {
+            throw new StorageException(String.format("There are no any items with ID %d!", itemId), HttpStatus.NOT_FOUND);
```
Нарушение требований кодстайла: Очень длинная строка - превышает 120 символов в длину.

IgorMartynkin1981 replied:
Выполнил перенос строки

kukolev commented on src/main/java/ru/practicum/shareit/user/User.java:
```diff
@@ -1,7 +1,17 @@
+import lombok.Data;
+
 /**
  * TODO Sprint add-controllers.
```
Это TODO нужно удалить, т.к. функционал реализован. Либо нужно написать что осталось реализовать.

IgorMartynkin1981 replied:
Заметки убрал

kukolev commented on src/main/java/ru/practicum/shareit/user/UserController.java:
```diff
@@ -1,12 +1,53 @@
+import java.util.Collection;
 
 /**
  * TODO Sprint add-controllers.
```
Это TODO нужно удалить, т.к. функционал реализован. Либо нужно написать что осталось реализовать.

IgorMartynkin1981 replied:
Заметки убрал

kukolev commented on src/main/java/ru/practicum/shareit/user/UserController.java:
```diff
@@ -1,12 +1,53 @@
+
+    @PatchMapping("/{id}")
+    public ResponseEntity<UserDto> updateUser(@PathVariable Long id, @Validated @RequestBody UserDto user, BindingResult errors) throws ValidationException {
+        if (errors.hasErrors()) throw new ValidationException(errors, HttpStatus.BAD_REQUEST);
```
Бизнес-логика должна быть реализована в сервисах, а не в контроллере. Именно сервис инкапсулирует всю логику работы приложения и выставляет наружу методы, которые потом вызываются из контроллеров или из других сервисов.

Аналогично нужно поправить ниже.

IgorMartynkin1981 replied:
Логика перенесена в сервис

kukolev commented on src/main/java/ru/practicum/shareit/user/UserServiceImpl.java:
```diff
@@ -0,0 +1,87 @@
+            }
+
+        } else {
+            throw new ValidationException(String.format("User with email %s already exists!", user.getEmail()), HttpStatus.CONFLICT);
```
Нарушение требований кодстайла: Очень длинная строка - превышает 120 символов в длину.

IgorMartynkin1981 replied:
Выполнил перенос строки

kukolev commented on src/main/java/ru/practicum/shareit/user/UserServiceImpl.java:
```diff
@@ -0,0 +1,87 @@
+            if (emailIsValid(userDto.getEmail())) {
+                builder.email(userDto.getEmail());
+            } else {
+                throw new ValidationException(String.format("User with email %s already exists!", user.getEmail()), HttpStatus.CONFLICT);
```
Нарушение требований кодстайла: Очень длинная строка - превышает 120 символов в длину.

IgorMartynkin1981 replied:
Выполнил перенос строки

IgorMartynkin1981 committed:
Refactor согласно замечаний №1:
1. delete лишние комментарии
2. длину строк привёл в соответствие стилю
3. перенёс логику из контроллера в сервис

IgorMartynkin1981 committed:
Refactor согласно замечаний №1:
1. delete лишние комментарии
2. длину строк привёл в соответствие стилю
3. перенёс логику из контроллера в сервис

kukolev commented on src/main/java/ru/practicum/shareit/user/UserServiceImpl.java:
```diff
@@ -0,0 +1,78 @@
+        userDAO.delete(userId);
+    }
+
+    private boolean emailIsValid(String email) {
```
Нарушение кодстайла: Название метода лучше начинать с глагола, обозначающего действие над аргументами или над классом, которому принадлежит метод. Так назначение метода будет понятно сразу из названия. Нужно поправить.

IgorMartynkin1981 replied:
Изменил имя метода на "checkUserForValid"

kukolev commented on src/main/java/ru/practicum/shareit/item/ItemServiceImpl.java:
```diff
@@ -0,0 +1,96 @@
+                .collect(Collectors.toList());
+    }
+
+    private boolean itemIsValid(ItemDto item) {
```
Нарушение кодстайла: Название метода лучше начинать с глагола, обозначающего действие над аргументами или над классом, которому принадлежит метод. Так назначение метода будет понятно сразу из названия. Нужно поправить.

IgorMartynkin1981 replied:
Изменил название метода на "checkItemForValid"

IgorMartynkin1981 committed:
Refactor согласно замечаний №2:
1. Изменил имя методов
